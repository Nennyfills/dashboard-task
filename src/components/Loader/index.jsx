import PropTypes from 'prop-types';

const Loader = ({ cols, rows, notTable }) => {
  const shimmerRows = [];
  for (let count = 1; count <= rows; count += 1) {
    const shimmerCols = [];
    for (let countC = 1; countC <= cols; countC += 1) {
      shimmerCols.push(
        <td key={countC} data-testid="col">
          <div key={`shimmer-${countC}`} className="shimmer">
            <div className="shimmer-line" />
          </div>
        </td>
      );
    }
    shimmerRows.push(
      <tr key={count} data-testid="row">{shimmerCols && shimmerCols.map((col) => col)}</tr>
    );
  }
  if (notTable) {
    return (
      <div className="shimmer" data-testid="single-loader">
        <div className="shimmer-line" />
      </div>
    );
  }
  return shimmerRows && shimmerRows.map((shim) => shim);
};

Loader.propTypes = {
  rows: PropTypes.number,
  cols: PropTypes.number,
  notTable: PropTypes.bool,
};

Loader.defaultProps = {
  rows: 4,
  cols: 4,
  notTable: false,
};

export default Loader;
