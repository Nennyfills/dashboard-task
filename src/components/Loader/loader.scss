.center-loader {
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  text-align: center;
  width: 100%;
}

.shimmer {
  animation-duration: 2.2s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: shimmer;
  animation-timing-function: linear;
  background: linear-gradient(
    to right,
    $light-grey-2 4%,
    #e0ecf3 25%,
    $light-grey 36%
  );
  background-size: 1200px 100%;
  border-radius: 4em;
  &-line {
    width: 100%;
    height: 15px;
    align-self: center;
    margin-left: 16px;
  }
}
@-webkit-keyframes shimmer {
  0% {
    background-position: -100% 0;
  }
  100% {
    background-position: 100% 0;
  }
}

@keyframes shimmer {
  0% {
    background-position: -1200px 0;
  }
  100% {
    background-position: 1200px 0;
  }
}
