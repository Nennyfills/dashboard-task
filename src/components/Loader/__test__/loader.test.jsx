import { render, screen } from '@testing-library/react';
import Loader from '..';

it('should take a snapshot', () => {
  const { asFragment } = render(<Loader />);
  expect(asFragment(<Loader />)).toMatchSnapshot();
});

it('Should render a Loader with test id single-loader', () => {
  render(<Loader cols={5} rows={7} notTable />);
  expect(screen.queryByTestId('single-loader')).toBeTruthy();
  expect(screen.queryAllByTestId('single-loader')).toHaveLength(1);
});

it('Should render a Loader with test id col', () => {
  render(<Loader cols={2} rows={3} />);
  expect(screen.queryAllByTestId('col')).toBeTruthy();
  expect(screen.queryAllByTestId('col')).toHaveLength(6);
});
