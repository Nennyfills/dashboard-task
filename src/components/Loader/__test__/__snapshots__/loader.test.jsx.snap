// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
  <div
    class="shimmer"
  >
    <div
      class="shimmer-line"
    />
  </div>
</DocumentFragment>
`;
